### DISCUSSION TITLE
用不了啊我小白

### DISCUSSION THREAD

**Discussion by shuilaijiujiuwo** [UnblockNeteaseMusic/luci-app-unblockneteasemusic#148] [General]
[](url
![V`~ ~AP(M8@VPTAP}{I)B3U](https://user-images.githubusercontent.com/105039642/167081077-6d5fcbbb-0324-4723-87ef-6d6276fbac9c.png)
![Q0XI$)OOZPW(ZVMF()_72 E](https://user-images.githubusercontent.com/105039642/167081081-9f75a9a1-785a-4245-8671-06da1c2ce777.png)
)

**Comment by shuilaijiujiuwo** [comment]
谁来教教我为什么我用不了啊

**Comment by 1715173329** [comment]
命令行执行 `unm-debug` 打印调试报告。

**Comment by ximiTech** [comment]
我也用不了，请帮忙看看。

root@OpenWrt:~# unm-debug
Launching luci-app-unblockneteasmusic Debugging Tool...


OpenWrt info:
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='SNAPSHOT'
DISTRIB_TARGET='x86/64'
DISTRIB_ARCH='x86_64'
DISTRIB_TAINTS='no-all'
DISTRIB_REVISION='R22.5.5'
DISTRIB_DESCRIPTION='OpenWrt '


uclient-fetch info:
Package: uclient-fetch
Version: 2021-05-14-6a6011df-1
Depends: libc, libuclient20201210
Provides: wget
Status: install user installed
Architecture: x86_64
Installed-Time: 1652243935

Package: libustream-openssl20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20210516, libopenssl1.1
Provides: libustream-openssl
Status: install user installed
Architecture: x86_64
Installed-Time: 1652243935

Downloading 'https://api.github.com/repos/UnblockNeteaseMusic/server/commits?sha=enhanced&path=precompiled'
Connecting to 13.230.158.120:443
Writing to stdout

Download completed (125581 bytes)
5fdfda6e606550d0dbfda12142f638235a3bd3ef


Node.js info:
Package: node
Version: v14.18.3-1
Depends: libc, libstdcpp6, libopenssl1.1, zlib, libnghttp2-14, libatomic1
Status: install user installed
Architecture: x86_64
Installed-Time: 1652243935

Node.js is placed at /usr/bin/node
Node.js version: v14.18.3


luci-app-unblockneteasmusic info:
Package: luci-app-unblockneteasemusic
Version: 2.12-4
Depends: libc, busybox, dnsmasq-full, ipset, jsonfilter, node, uclient-fetch
Status: install user installed
Architecture: all
Conffiles:
 /etc/config/unblockneteasemusic e9413f9ce065cfb7b875d1948c6d3258772ab8854b7ccf193df49c3c8b6cca20
Installed-Time: 1652323380

-rw-r--r--    1 root     root         255 May 12 10:53 /etc/config/unblockneteasemusic
-rwxr-xr-x    1 root     root       10.6K Apr 25 08:21 /etc/init.d/unblockneteasemusic

/usr/share/unblockneteasemusic:
drwxr-xr-x    2 root     root        3.4K May 12 10:45 core
-rw-r--r--    1 root     root          41 May 12 10:45 core_local_ver
-rwxr-xr-x    1 root     root        5.5K Apr 25 08:21 debugging.sh
-rwxr-xr-x    1 root     root         348 Apr 25 08:21 log_check.sh
drwxr-xr-x    2 root     root        3.4K May 12 10:43 rules
-rwxr-xr-x    1 root     root        2.9K Apr 25 08:21 update.sh

config unblockneteasemusic 'config'
        option replace_music_source 'dont_replace'
        option use_custom_cookie '0'
        option local_vip '0'
        option enable '1'
        option music_source 'kuwo'
        option enable_flac '1'
        option auto_update '0'
        option advanced_mode '0'



UnblockNeteaseMusic Node.js info:
Git HEAD version: 5fdfda6e606550d0dbfda12142f638235a3bd3ef
Core version: 0.27.0-rc.6
-rw-r--r--    1 root     root      472.8K May 12 10:45 app.js
-rw-r--r--    1 root     root      416.2K May 12 10:45 bridge.js
-rw-r--r--    1 root     root         790 May 12 10:45 ca.crt
-rw-r--r--    1 root     root         843 May 12 10:45 server.crt
-rw-r--r--    1 root     root         359 May 12 10:45 server.key


Netease networking info:
< Location: http://m10.music.126.net/20220512112731/e47ffe60f1665010c103547bc7914152/ymusic/obj/w5zDlMODwrDDiGjCn8Ky/3057940231/3d32/93a3/4db1/0757f3a378b41e09ca2767c1abd47afa.mp3
{"timestamp":1652324684458,"status":404,"error":"Not Found","message":"No message available","path":"/httpdns/v2/d"}

Port status:
tcp        0      0 0.0.0.0:5200            0.0.0.0:*               LISTEN      26163/node
tcp        0      0 0.0.0.0:5201            0.0.0.0:*               LISTEN      26163/node


Running info:
PROCD running status: running
26163 root      243m S    node /usr/share/unblockneteasemusic/core/app.js -a 0.0.0.0 -p 5200:5201 -o kuwo -e https://music.163.com


Firewall info:
Chain netease_cloud_music (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             0.0.0.0/8
RETURN     all  --  anywhere             10.0.0.0/8
RETURN     all  --  anywhere             127.0.0.0/8
RETURN     all  --  anywhere             169.254.0.0/16
RETURN     all  --  anywhere             172.16.0.0/12
RETURN     all  --  anywhere             192.168.0.0/16
RETURN     all  --  anywhere             base-address.mcast.net/4
RETURN     all  --  anywhere             240.0.0.0/4
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_http src tcp dpt:www redir ports 5200
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_https src tcp dpt:https redir ports 5201

Name: neteasemusic
Type: hash:ip
Revision: 4
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 736
References: 1
Number of entries: 11
Members:
59.111.181.38
115.236.121.1
59.111.160.197
59.111.181.60
193.112.159.225
118.24.63.156
59.111.181.35
223.252.199.66
223.252.199.67
115.236.118.33
59.111.160.195

Name: acl_neteasemusic_http
Type: hash:ip
Revision: 4
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 208
References: 1
Number of entries: 0
Members:

Name: acl_neteasemusic_https
Type: hash:ip
Revision: 4
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 208
References: 1
Number of entries: 0
Members:

dhcp-option=252,http://10.10.10.252:5200/proxy.pac
ipset=/.music.163.com/neteasemusic
ipset=/interface.music.163.com/neteasemusic
ipset=/interface3.music.163.com/neteasemusic
ipset=/apm.music.163.com/neteasemusic
ipset=/apm3.music.163.com/neteasemusic
ipset=/clientlog.music.163.com/neteasemusic
ipset=/clientlog3.music.163.com/neteasemusic


Testing source replacing...
{"data":[{"id":33190502,"url":null,"br":0,"size":0,"md5":null,"code":404,"expi":1200,"type":null,"gain":0.0,"fee":0,"uf":null,"payed":0,"flag":257,"canExtend":false,"freeTrialInfo":null,"level":null,"encodeType":null,"freeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"listenType":null},"freeTimeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"type":0,"remainTime":0},"urlSource":0}],"code":200}
{"data":[{"id":33190502,"url":"https://music.163.com/package/aHR0cDovL2VwLnN5Y2RuLmt1d28uY24vYTA3YjczM2YxMTMyMDVhMjllNDgwMzA2ODc4NWVhNzQvNjI3Yzc4YzcvcmVzb3VyY2UvczMvMC84NS8zMjcxMTI0MTY3LmZsYWM=/33190502.flac","br":999000,"size":30395622,"md5":"e17c5cb38fc839c860f1497d8798f4f2","code":200,"expi":1200,"type":"flac","gain":0,"fee":0,"uf":null,"payed":0,"flag":0,"canExtend":false,"freeTrialInfo":null,"level":null,"encodeType":null,"freeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"listenType":null},"freeTimeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"type":0,"remainTime":0},"urlSource":0}],"code":200}
{"level":30,"time":1652324552091,"pid":26163,"hostname":"OpenWrt","scope":"provider/match","audioId":33190502,"songName":"君のままで","url":"http://ep.sycdn.kuwo.cn/a07b733f113205a29e4803068785ea74/627c78c7/resource/s3/0/85/3271124167.flac","msg":"Replaced: [33190502] 君のままで"}


Log is available at:
https://litter.catbox.moe/s8y7mk.txt
https://transfer.sh/7LBiwd/unm-debugging-output.txt

**Comment by ximiTech** [comment]
{"level":30,"time":1652327226374,"pid":20021,"hostname":"OpenWrt","scope":"app","msg":"HTTP Server running @ http://0.0.0.0:5200"}
{"level":30,"time":1652327226375,"pid":20021,"hostname":"OpenWrt","scope":"app","msg":"HTTPS Server running @ http://0.0.0.0:5201"}

**Comment by ximiTech** [comment]
我发现了，云音乐客户端版本不要超过2.9.8则可以使用，客户端最新版现在到了2.10就不行了。

**Reply by heaven-1314** [reply]
正解

**Comment by 1715173329** [comment]
建议用旧版 2.5.0
